contribs/gnodev/pkg/dev/node_test.go

```diff
@@ -0,0 +1,306 @@
+package dev
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gnolang/gno/contribs/gnodev/pkg/emitter"
+	"github.com/gnolang/gno/contribs/gnodev/pkg/events"
+	"github.com/gnolang/gno/gno.land/pkg/gnoclient"
+	"github.com/gnolang/gno/gno.land/pkg/integration"
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	core_types "github.com/gnolang/gno/tm2/pkg/bft/rpc/core/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// XXX: We should probably use txtar to test this package
```

Comment:
I think it's cool we have unit tests for this functionality, doing txtar captures would be a nightmare

Reply:
![el-dorado-both](https://github.com/gnolang/gno/assets/8671905/3e1ddb4b-6db5-4b63-bad4-26d055340afb)

in this current state it's not really doable anyway, but I plan to make txtar setup modular so you can easily add other tool like `gnodev` .

Unit tests are definitely something we want to keep, but having both would definitely help create a more complex flow that is not easily achievable with just unit tests.